File: benchmarks/pandas/bench_merge_left_on_right_on.py
"""Benchmark: merge with left_on/right_on (pandas equivalent)."""
import json
import time
import pandas as pd

ROWS = 20_000
WARMUP = 3
ITERATIONS = 10

left = pd.DataFrame({
    "emp_id": list(range(ROWS)),
    "salary": [30000 + i * 10 for i in range(ROWS)],
})
right = pd.DataFrame({
    "id": list(range(ROWS // 2)),
    "dept": [f"dept{i % 10}" for i in range(ROWS // 2)],
})

for _ in range(WARMUP):
    pd.merge(left, right, left_on="emp_id", right_on="id")

t0 = time.perf_counter()
for _ in range(ITERATIONS):
    pd.merge(left, right, left_on="emp_id", right_on="id")
total = (time.perf_counter() - t0) * 1000

print(json.dumps({"function": "merge_left_on_right_on", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_merge_left_on_right_on.ts
/**
 * Benchmark: merge with left_on/right_on — join on differently-named columns.
 */
import { DataFrame, merge } from "../../src/index.js";

const ROWS = 20_000;
const WARMUP = 3;
const ITERATIONS = 10;

const left = DataFrame.fromColumns({
  emp_id: Array.from({ length: ROWS }, (_, i) => i),
  salary: Array.from({ length: ROWS }, (_, i) => 30000 + i * 10),
});
const right = DataFrame.fromColumns({
  id: Array.from({ length: ROWS / 2 }, (_, i) => i),
  dept: Array.from({ length: ROWS / 2 }, (_, i) => `dept${i % 10}`),
});

for (let i = 0; i < WARMUP; i++) {
  merge(left, right, { left_on: "emp_id", right_on: "id" });
}

const t0 = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  merge(left, right, { left_on: "emp_id", right_on: "id" });
}
const total = performance.now() - t0;

console.log(
  JSON.stringify({ function: "merge_left_on_right_on", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }),
);
